src/java/org/apache/cassandra/db/rows/BufferCell.java

```diff
@@ -116,15 +117,15 @@ public CellPath path()
 
     public Cell<?> withUpdatedColumn(ColumnMetadata newColumn)
     {
-        return new BufferCell(newColumn, timestamp, ttl, localDeletionTime, value, path);
+        return new BufferCell(newColumn, timestamp, ttl, deletionTimeGuavaUintToLong(localDeletionTimeGuavaUint), value, path);
```

Comment:
Same as in `ArrayCell`, we can use a private constructor taking the uint so we don't need to do the transformation twice. Same in the method right below.